lib/Controller/OutOfOfficeController.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Controller;
+
+use DateTimeImmutable;
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Http\JsonResponse;
+use OCA\Mail\Http\TrapError;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Service\OutOfOffice\OutOfOfficeState;
+use OCA\Mail\Service\OutOfOfficeService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\IRequest;
+use OCP\IUserSession;
+use OCP\User\IAvailabilityCoordinator;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+
+class OutOfOfficeController extends Controller {
+	private ?IAvailabilityCoordinator $availabilityCoordinator;
+
+	public function __construct(
+		IRequest $request,
+		ContainerInterface $container,
+		private IUserSession $userSession,
+		private AccountService $accountService,
+		private OutOfOfficeService $outOfOfficeService,
+		private ITimeFactory $timeFactory,
+	) {
+		parent::__construct(Application::APP_ID, $request);
+
+		try {
+			$this->availabilityCoordinator = $container->get(IAvailabilityCoordinator::class);
```

Comment:
you can add a note that with 28+ we can use simpler syntax: https://github.com/nextcloud/server/pull/41265